lib/Conch/Route/Device.pm

```diff
@@ -14,128 +14,496 @@ Conch::Route::Device
+
+=back
+
+=cut
+
     # POST /device/:device_id
     $device->post('/:device_id')->to('device_report#process');
 
+=head3 C<GET /device?key=:value>
+
+Looks up one or more devices by query parameter. Supports:
+
+    /device?hostname=$hostname
+    /device?mac=$macaddr
+    /device?ipaddr=$ipaddr
+    /device?$setting_key=$setting_value
+
+Returns a C<403 FORBIDDEN> if the current user doesn't have permissions on the
+device through at least one workspace.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<Devices|https://conch.joyent.us/schema/response/Devices>
+
+=back
+
+=cut
+
     # GET /device?key=:value
     $device->get('/')->to('device#lookup_by_other_attribute');
 
     {
         # chainable action that extracts and looks up device_id from the path
         my $with_device = $device->under('/:device_id')->to('device#find_device');
 
+=head3 C<GET /device/:device_id>
+
+Retrieves details about a single (active) device.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DetailedDevice|https://conch.joyent.us/schema/response/DetailedDevice>
+
+=back
+
+=cut
+
         # GET /device/:device_id
         $with_device->get('/')->to('device#get');
 
+=head3 C<GET /device/:device_id/pxe>
+
+Gets PXE-specific information about a device.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DevicePXE|https://conch.joyent.us/schema/response/DevicePXE>
+
+=back
+
+=cut
+
         # GET /device/:device_id/pxe
         $with_device->get('/pxe')->to('device#get_pxe');
 
+=head3 C<GET /device/:device_id/phase>
+
+Gets just the device's phase.
+
+=over 4
+
+=item * Requires Authentication
+
+=item * Response Schema: L<DevicePhase|https://conch.joyent.us/schema/response/DevicePhase>
+
+=back
+
+=cut
+
         # GET /device/:device_id/phase
         $with_device->get('/phase')->to('device#get_phase');
 
+=head3 C<POST /device/:device_id/graduate>
+
+Marks the device as "graduated" (VLAN flipped). Returns C<409 CONFLICT> if the
+device has already been graduated. Returns C<303 SEE OTHER> with the URL for
+the device upon success.
+
+=over 4
+
+=item * Requires Authentication
+
+=back
+
+=cut
+
         # POST /device/:device_id/graduate
         $with_device->post('/graduate')->to('device#graduate');
+
+=head3 C<POST /device/:device_id/triton_setup>
+
+If a device has been marked as rebooted into Triton and has a Triton UUID, sets
+the C<triton_setup> field. Returns a C<409 CONFLICT> if the device has not yet
+been rebooted and assigned a UUID or has already been marked. Upon success
+returns C<303 SEE OTHER> with the URL of the device.
+
+=over 4
+
+=item * Requires Authentication
+
+=back
+
+=cut
+
         # POST /device/:device_id/triton_setup
         $with_device->post('/triton_setup')->to('device#set_triton_setup');
+
+=head3 C<POST /device/:device_id/triton_uuid>
+
+Sets the C<triton_uuid> field on a device, given a triton_uuid field that is a
```

Comment:
This is a text retelling of the JSON schema for the request. Let the schema say this for itself

Reply:
Should be removed in favor of the link to the schema.